mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -0,0 +1,734 @@
+    )
+    .await;
+    if base_vwap == calculated_price && rel_vwap == calculated_price {
+        return calculated_price;
+    }
+    let mut to_divide = 0;
+    let mut total_vwap = MmNumber::default();
+    if !base_swaps_empty {
+        to_divide += 1;
+        total_vwap += base_vwap;
+    }
+    if !rel_swaps_empty {
+        to_divide += 1;
+        total_vwap += rel_vwap;
+    }
+    // here divide cannot be 0 anymore because if both swaps history are empty we do not pass through this function.
+    let vwap_price = total_vwap / MmNumber::from(to_divide);
+    if vwap_price > calculated_price {
+        info!(
+            "[{}/{}]: price: {} is less than average trading price ({} swaps): - using vwap price: {}",
+            cfg.base, cfg.rel, calculated_price, nb_valid_trades, vwap_price
+        );
+        return vwap_price;
+    }
+    info!("price calculated by the CEX rates {} is above the vwap price ({} swaps) {} - skipping threshold readjustment for pair: [{}/{}]", 
+            calculated_price, nb_valid_trades, vwap_price, cfg.base, cfg.rel);
+    calculated_price
+}
+
+pub async fn vwap(
+    base_swaps: MyRecentSwapsResponse,
+    rel_swaps: MyRecentSwapsResponse,
+    calculated_price: MmNumber,
+    cfg: &SimpleCoinMarketMakerCfg,
+) -> MmNumber {
+    // since the limit is `1000` unwrap is fine here.
+    let nb_diff_swaps = rel_swaps.swaps.len().to_isize().unwrap() - base_swaps.swaps.len().to_isize().unwrap();
+    let have_precedent_swaps = !rel_swaps.swaps.is_empty() && !base_swaps.swaps.is_empty();
+    if nb_diff_swaps.is_zero() && !have_precedent_swaps {
+        info!(
+            "No last trade for trading pair: [{}/{}] - keeping calculated price: {}",
+            cfg.base, cfg.rel, calculated_price
+        );
+        return calculated_price;
+    }
+    vwap_logic(base_swaps, rel_swaps, calculated_price, cfg).await
+}
+
+async fn vwap_calculator(
+    calculated_price: MmNumber,
+    ctx: &MmArc,
+    cfg: &SimpleCoinMarketMakerCfg,
+) -> VwapProcessingResult {
+    let my_recent_swaps_req = async move |base: String, rel: String| MyRecentSwapsReq {
+        paging_options: PagingOptions {
+            limit: 1000,
+            page_number: NonZeroUsize::new(1).unwrap(),
+            from_uuid: None,
+        },
+        filter: MySwapsFilter {
+            my_coin: Some(base),
+            other_coin: Some(rel),
+            from_timestamp: None,
+            to_timestamp: None,
+        },
+    };
+    let base_swaps = my_recent_swaps(
+        ctx.clone(),
+        my_recent_swaps_req(cfg.base.clone(), cfg.rel.clone()).await,
+    )
+    .await?;
+    let rel_swaps = my_recent_swaps(
+        ctx.clone(),
+        my_recent_swaps_req(cfg.rel.clone(), cfg.base.clone()).await,
+    )
+    .await?;
+    Ok(vwap(base_swaps, rel_swaps, calculated_price, cfg).await)
+}
+
+async fn cancel_pending_orders(ctx: &MmArc) {
+    match cancel_all_orders(ctx.clone(), CancelBy::All).await {
```

Comment:
It cancels all orders including ones, which were created manually. Could you please refactor it to cancel only for pairs managed by the bot?